File: Structural/Decorator/Python/milk_decorator.py
# milk_decorator.py

from coffee_decorator import CoffeeDecorator

class MilkDecorator(CoffeeDecorator):
  def get_description(self) -> str:
    return self._decorated_coffee.get_description() + ", Milk"

  def get_cost(self) -> float:
    return self._decorated_coffee.get_cost() + 1.5

File: Structural/Decorator/JavaScript/milkDecorator.js
// milkDecorator.js

const CoffeeDecorator = require("./coffeeDecorator");

class MilkDecorator extends CoffeeDecorator {
  getDescription() {
    return this.decoratedCoffee.getDescription() + ", Milk";
  }

  getCost() {
    return this.decoratedCoffee.getCost() + 1.5;
  }
}

module.exports = MilkDecorator;
